zilencer/views.py

```diff
@@ -153,23 +154,35 @@ def register_remote_server(
 def register_remote_push_device(
     request: HttpRequest,
     entity: Union[UserProfile, RemoteZulipServer],
-    user_id: int = REQ(json_validator=check_int),
+    user_id: Optional[int] = REQ(json_validator=check_int, default=None),
+    user_uuid: Optional[str] = REQ(default=None),
     token: str = REQ(),
     token_kind: int = REQ(json_validator=check_int),
     ios_app_id: Optional[str] = None,
 ) -> HttpResponse:
     server = validate_bouncer_token_request(entity, token, token_kind)
 
+    if not (user_id or user_uuid):
+        raise JsonableError(_("Missing user_id or user_uuid"))
+    if user_id and user_uuid:
+        raise JsonableError(_("Specify only one of user_id or user_uuid"))
```

Comment:
Why does the underlying UserPushIndentityCompat support having both, when we don't accept both at the interface?

It seems cleaner to me to have UserPushIndentityCompat to be user_id `xor` user_uuid, not `or`.

----

Oh, is the case for taking both that it makes the removal easier for the client?

I guess the added complexity is probably worth the simplicity of the removal being able to make only one request.  It may be worth commenting here and/or on the UserPushIndentityCompat class that we have different models for or vs xor based on where it's being used from.

Reply:
@alexmv The idea here is that:
1. This is the "register" endpoint. On other endpoints, we receive both `user_id` and `user_uuid` value from the remote server - so that we can handle *old* token which were registered with `user_id`. E.g. when we receive "send notification to this user" request, we want to grab `RemotePushToken`s matching either the `user_uuid` OR the `user_id` (or the notification wouldn't be sent to the user's devices that registered before 5.0). 
2. On this "register" endpoint, we don't worry about back-compatibility like that - actually, it would probably be wrong to accept a registration with both `user_id` and `user_uuid` - we want to have registration of just one, clear type, Either UUID (post-5.0 servers) or ID (pre-5.0 servers).

Does that make sense?